### ISSUE TITLE
Setting query server definition

### ISSUE THREAD

**Issue by endyjasmi** [apache/couchdb-docker#149]
## Expected Behavior
According to https://docs.couchdb.org/en/latest/config/query-servers.html#query-servers , I should be able to set query server parameter with environment variable.

## Current Behavior
When I set the environment variable `COUCHDB_QUERY_SERVER_JAVASCRIPT="/opt/couchdb/bin/couchjs --eval /opt/couchdb/share/server/main.js"` doesn't seems to take effect.

## Possible Solution
Currently I am mounting over the default `/opt/couchdb/bin/couchdb` binary file and setting it directly at that file. This however sometime causes permissions problem.

## Context
Basically I am currently using AJV for JSON schema validation purposes and a few other library (lodash, crytpojs, etc...) for other purposes. In other for this library to works, I'll need to turn eval flag on in couchjs. Prior to version 2.3, this query server definition is set inside the INI file but have since changed to environment variable.

## Your Environment
* Version used: CouchDB 2.3.1
* Operating System and version (desktop or mobile): Windows 10 and Other linux environment

**Comment by wohali** [comment]
How are you passing the environment variable into Docker? Are you using the `-e` argument and it still didn't work?

**Comment by wohali** [comment]
Hi @endyjasmi ,

This is actually a CouchDB bug in `/opt/couchdb/bin/couchdb`, not in Docker, as that startup script isn't written to allow those variables to be overridden by the calling process. I've opened a new issue in CouchDB to track this bug:

When the bug is fixed there, it'll be fixed here.

Another fix you could consider is to define a new language, like `docker run -e COUCHDB_QUERY_SERVER_MYJAVASCRIPT="...." apache/couchdb:2.3.1` then change your `language` setting in view files to `myjavascript`. I tested this approach and it works fine.
